examples/cpp/helloworld/greeter_server_ssl.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <grpcpp/ext/proto_server_reflection_plugin.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/health_check_service_interface.h>
+
+#ifdef BAZEL_BUILD
+#include "examples/protos/helloworld.grpc.pb.h"
+#else
+#include "helloworld.grpc.pb.h"
+#endif
+#include "greeter_utils.h"
+
+using ::grpc::Server;
+using ::grpc::ServerBuilder;
+using ::grpc::ServerContext;
+using ::grpc::Status;
+
+using ::helloworld::Greeter;
+using ::helloworld::HelloReply;
+using ::helloworld::HelloRequest;
+
+// Logic and data behind the server's behavior.
+class GreeterServiceImpl final : public Greeter::Service {
+  Status SayHello(ServerContext* context, const HelloRequest* request,
+                  HelloReply* reply) override {
+    std::string prefix{"Hello "};
+
+    reply->set_message(prefix + request->name());
+
+    return Status::OK;
+  }
+};
+
+void run_server() {
+  std::string server_address{"localhost:50051"};
+
+  grpc::EnableDefaultHealthCheckService(true);
+  grpc::reflection::InitProtoReflectionServerBuilderPlugin();
+
+  ServerBuilder builder;
+
+  // Use the gen_certs.sh for the generation of required certificates
+  // [!] Be carefull here using a server.crt with the CN != localhost [!]
+  std::string key, cert, root;
+
+  read("server.crt", cert);
+  read("server.key", key);
+  read("ca.crt", root);
+
+  // Configure SSL options
+  grpc::SslServerCredentialsOptions::PemKeyCertPair keycert = {key, cert};
+  grpc::SslServerCredentialsOptions sslOps;
```

Comment:
The `Ssl*Credentials` is the old credential type we support. It's good to have them in our examples, but for any new features we added to our security API, we'd add it to `Tls*Credentials`.
Are you interested in adding `Tls*Credentials` example as well? It would be very similar to this one. As a reference for its usage, here is the e2e test for `Tls*Credentials`: https://github.com/grpc/grpc/pull/28456(not merged yet, but likely it will soon)

Reply:
I saw that Tls*Credentials is still in experimental namespace. I will update this examples when this change.